src/dotnet/CarbonAware/IntervalHelper.cs

```diff
@@ -0,0 +1,40 @@
+namespace CarbonAware;
+
+/// <summary>
+/// Helper for intervals within Carbon Aware.
+/// </summary>
+public static class IntervalHelper
+{
+
+    /// <summary>
+    /// Ensures the data is available between the range of time considering the duration of the data
+    /// </summary>
+    /// <param name="expandedData">Expanded data that includes the minimum sampling window</param>
+    /// <param name="startDate">Original start date provided by user</param>
+    /// <param name="endDate">Original end date provided by user</param>
+    /// <returns>Filtered emissions data.</returns>
+    public static IEnumerable<EmissionsData> MinSamplingFiltering(IEnumerable<EmissionsData> expandedData, DateTimeOffset startDate, DateTimeOffset endDate, TimeSpan duration = default)
```

Comment:
NIT: Both of the method names here are the original names from before the internal logic evolved.  Worth considering if those are still the names you want.  EG `MinSamplingFiltering` used to care more about the `MinSamplingWindow` but this method doesn't really and potentially has broader use.